src/compiler/output-targets/output-utils.ts

```diff
@@ -153,6 +117,18 @@ export const VALID_TYPES = [
   COPY,
   CUSTOM,
   STATS,
-];
+] as const;
+
+type ValidConfigOutputTarget = typeof VALID_CONFIG_OUTPUT_TARGETS[number];
+
+/**
+ * Check whether a given output target is a valid one to be set in a Stencil config
+ */
+export function isValidConfigOutputTarget(targetType: string): targetType is ValidConfigOutputTarget {
+  // unfortunately `includes` is typed on `Array<T>` as `(el: T):
+  // boolean` so a `string` cannot be passed to `includes` on a
+  // `ReadonlyArray` 😢 thus we `as any`
```

Comment:
This is a bit of an annoying thing with typescript. If you have a `ReadonlyArray` like this:

```ts
const xs = ["foo", "bar"] as const;
```

then `type arrEntry = xs[number]` will be of type `"foo" | "bar"`, which is great! But unfortunately, array methods are also all typed like this. You of course want this for things like `Array.prototype.map`, that should be something like `map<T>: (x: "foo" | "bar") => T`, but we don't really want this for `Array.prototype.includes`. The problem is that `ReadonlyArray.prototype.includes` also has the type of the array entries, so it's something like

```ts
interface ReadOnlyArray {
  includes: (x: "foo" | "bar") => boolean
}
```

I think I'd prefer 

```ts
interface ReadOnlyArray {
  includes: (x: string) => x is "foo" | "bar"
}
```

since narrowing the type of the argument for `.includes` means that I can't take an arbitrary string and test if it's found in a `ReadonlyArray` I've declared 😢 

I found an issue on the typescript repo talking about this and discussing various workarounds: https://github.com/microsoft/TypeScript/issues/31018

Anywhoo - to work around this I added this typeguard function here which, while it has an internal type hole (`as any`), does let us have a way to check for membership within a `ReadonlyArray` for an arbitrary string. Not sure this is the best solution though!